### PR TITLE
add ST sequence, fix emacs

### LINKED ISSUE #3700: Emacs xterm specific initialization does not complete properly
The emacs cannot define xterm configuration of the background color.
I've debugged the issue down to the point when [emacs queries the xterm background color](https://github.com/emacs-mirror/emacs/blob/emacs-27/lisp/term/xterm.el#L737-L738). 

Unfortunately, I cannot provide more exact reasons and errors due to a lack of emacs debugging experience.

## Details
- Browser and browser version: Electron 17
- OS version: macOS 12.3 
- xterm.js version: 4.17.0 and 4.18. Possibly, 4.16.0 affected, too.
- emacs: 26 or 27. Most probably, older versions are affected, too

### Steps to reproduce

1. Run the `emacs` command
2. Observe the current window
3. `*scratch*` is opened. Expected that the `*GNU Emacs*` window is opened.

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #3700 

https://user-images.githubusercontent.com/29464607/159740259-35468672-fdc2-4c47-8757-76c0f6428e17.mov

### CODE REVIEW

File: src/browser/Terminal.ts [hunk 1]
```diff
@@ -224,7 +224,7 @@ export class Terminal extends CoreTerminal implements ITerminal {
             const channels = color.toColorRGB(acc === 'ansi'
               ? this._colorManager.colors.ansi[req.index]
               : this._colorManager.colors[acc]);
-            this.coreService.triggerDataEvent(`${C0.ESC}]${ident};${toRgbString(channels)}${C0.BEL}`);
+            this.coreService.triggerDataEvent(`${C0.ESC}]${ident};${toRgbString(channels)}ESC \9c`);
```
Comment (Tyriar) [inline src/browser/Terminal.ts:227]: Think this is right:

```suggestion
            this.coreService.triggerDataEvent(`${C0.ESC}]${ident};${toRgbString(channels)}${C1.ST}`);
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/browser/Terminal.ts (hunk 1 above)
Comment (Tyriar) [reply src/browser/Terminal.ts:227]: Hmm, maybe I don't know how to use ESC notation 🤔

Comment (jerch) [reply src/browser/Terminal.ts:227]: Ah with `\` I meant the backslash ASCII char, which is at position 0x5c (or `\x5c` as hex-char). So ST would become `{C0.ESC}\x5c`, or `{C0.ESC}\\` (javascript needs double \\ to print one backslash). Maybe it is a good idea to create another convenient mapping for the escaped C1 chars in `common/data/EscapeSequences`, like this:
```typescript
const C1_escaped = {
  ST: `{C0.ESC}\\`,
  // add wotever other C1 is needed in escaped form
}
```
Then you can use it directly as `{C1_escaped.ST}` and so on.

The escaped form of C1 chars of interest can be formalized as: 
```typescript
function escape_char(c) {
  return '\x1b' + String.fromCharCode(c.charCodeAt(0) - 0x40)
}
```

**Review by Tyriar — APPROVED** [review]

**Review by jerch — APPROVED** [review]
LGTM :+1:  (tests still need to be adjusted)

File: test/api/InputHandler.api.ts [hunk 2]
```diff
@@ -399,36 +399,36 @@ describe('InputHandler Integration Tests', function(): void {
       });
       it('query single color', async () => {
         await writeSync(page, '\x1b]4;0;?\x07');
-        assert.deepEqual(await page.evaluate('window._recordedData'), ['\x1b]4;0;rgb:2e2e/3434/3636\x07']);
+        assert.deepEqual(await page.evaluate('window._recordedData'), ['\x1b]4;0;rgb:2e2e/3434/3636{C0.ESC}\\']);
```
Comment (jerch) [inline test/api/InputHandler.api.ts:402]: @meganrogge Those strings are no template strings, thus the {...} parts will not get eval'ed. Either make them template strings + importing `C0`, or use the hex-string notation as '\x1b\\'.

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: thanks, just realized that. What's surprising to me is that some of the tests are passing despite that 🤔

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: Oh I see that's because window.recordedData is a string

Screen Shot 2022-03-23 at 3 45 22 PM

Comment (jerch) [reply test/api/InputHandler.api.ts:402]: Hmm thats weird and should not happen. Idk - maybe "printf debugging" is your friend?

Comment (jerch) [reply test/api/InputHandler.api.ts:402]: Woops - isnt there a `$` sign missing in `C1_ESCAPED.ST`?

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: that was a good thought, but using the name space didn't help either 🤔 .

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: but that probably won't fix the issue that you mentioned for other encodings.

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: If using C0.ESC isn't an issue here, shouldn't this just use that same thing for the terminating escape?

https://github.com/meganrogge/xterm.js/blob/3457fb48726cea910a3c212bd4943ad360fe3f2e/src/browser/Terminal.ts#L227

Comment (jerch) [reply test/api/InputHandler.api.ts:402]: > Maybe I just need to change the expected to match this? Screen Shot 2022-03-23 at 4 49 05 PM

Yeah, maybe it works with that namespace stuff instead. But the template string should contain `${C0.ESC}\\` (note the double backslashes and the $ sign, which was missing above).

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: it gets interpreted as a literal
🤷🏼‍♀️ 
Screen Shot 2022-03-23 at 5 16 17 PM
Screen Shot 2022-03-23 at 5 17 59 PM

Comment (jerch) [reply test/api/InputHandler.api.ts:402]: Almost - there is one superfluous $ char :sweat_smile:

Comment (meganrogge) [reply test/api/InputHandler.api.ts:402]: 😅  I took this literally

> But the template string should contain ${C0.ESC}\\

Works now 👍🏼  thanks so much for the help

**Pull request merged**